hamilton/plugins/h_ray.py

```diff
@@ -184,7 +193,20 @@ def check_node_type_equivalence(node_type: typing.Type, input_type: typing.Type)
         return node_type == input_type
 
     def execute_node(self, node: node.Node, kwargs: typing.Dict[str, typing.Any]) -> typing.Any:
-        return ray.remote(raify(node.callable)).bind(**kwargs)
+        """Function that is called as we walk the graph to determine how to execute a hamilton function.
+
+        :param node: the node from the graph.
+        :param kwargs: the arguments that should be passed to it.
+        :return: returns a ray object reference.
+        """
+        # Tags are added to nodes via the @ray_remote_options decorator
+        ray_tags = {
+            tag_name: tag_value
+            for tag_name, tag_value in node.tags.items()
+            if tag_name.startswith(f"{RAY_REMOTE_TAG_NAMESPACE}.")
+        }
+        ray_options = {name.split(".", 1)[1]: json.loads(value) for name, value in ray_tags.items()}
```

Comment:
can we pull this out into a function to unit test it?

Otherwise I don't see an easy integration test (that would live here https://github.com/DAGWorks-Inc/hamilton/tree/main/plugin_tests/h_ray) for this?